src/engine.rs

```diff
@@ -109,6 +109,9 @@ pub struct Reedline {
     history_cursor_on_excluded: bool,
     input_mode: InputMode,
 
+    // ExecuteHostCommand
+    execute_host_command: bool,
+
```

Comment:
Small nit: This naming and comment goes a bit lost among all the other configuration focussed members of `Reedline`

It describes a temporary state of `Reedline` so maybe `executing_host_command`/`after_host_command` is more explicit. I don't have a strong opinion where it should be placed in the overall struct, but there we could also emphasize its role.

```suggestion
    // Yielded to the host program after a `ReedlineEvent::ExecuteHostCommand`, thus redraw in-place
    executing_host_command: bool,

```